### PR TITLE
docs: update "Transformation" Plugins 

### PR DESCRIPTION
**Pull request by hf400159**
### Description

Updates the documentation of the "Transformation" Plugins.

Fixes issue    [6833](https://github.com/apache/apisix/issues/6833.)

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [ ] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [x] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

File: docs/zh/latest/plugins/fault-injection.md [hunk 1]
```diff
@@ -23,23 +29,33 @@ title: fault-injection
 
 ## 描述
 
-故障注入插件，该插件可以和其他插件一起使用，并且会在其他插件前被执行，配置 `abort` 参数将直接返回给客户端指定的响应码并且终止其他插件的执行，配置 `delay` 参数将延迟某个请求，并且还会执行配置的其他插件。
+`fault-injection` 插件，该插件可以和其他插件一起使用，并且会在其他插件前被执行。
```
Comment (SylviaBABY) [inline docs/zh/latest/plugins/fault-injection.md:32]:
```suggestion
`fault-injection` 为故障注入插件，该插件可以和其他插件一起使用，并在其他插件前被执行。
```

File: docs/zh/latest/plugins/fault-injection.md [hunk 2]
```diff
@@ -23,23 +29,33 @@ title: fault-injection
...
+
+`abort` 属性将直接返回给客户端指定的响应码并且终止其他插件的执行。
+
+`delay` 属性将延迟某个请求，并且还会执行配置的其他插件。
```
Comment (SylviaBABY) [inline docs/zh/latest/plugins/fault-injection.md:36]:
```suggestion

```

File: docs/zh/latest/plugins/fault-injection.md [hunk 3]
```diff
@@ -23,23 +29,33 @@ title: fault-injection
...
+| abort.body        | string  | 否   |             | 返回给客户端的响应数据。支持使用 NGINX 变量，如 `client addr: $remote_addr\n`|
+| abort.percentage  | integer | 否   |  [0, 100]   | 将被中断的请求占比         |
+| abort.vars        | array[] | 否   |             | 执行故障注入的规则，当规则匹配通过后才会执行故障注。`vars` 是一个表达式的列表，来自 [lua-resty-expr](https://github.com/api7/lua-resty-expr#operator-list)。 |
+| delay.duration    | number  | 是   |             | 延迟时间，可以指定小数     |
+| delay.percentage  | integer | 否   |  [0, 100]   | 将被延迟的请求占比         |
+| delay.vars        | array[] | 否   |             | 执行请求延迟的规则，当规则匹配通过后才会延迟请求。`vars` 是一个表达式列表，来自 [lua-resty-expr](https://github.com/api7/lua-resty-expr#operator-list)。   |
+
+:::info IMPORTANT
+
+`abort` 和 `delay` 属性至少要配置一个。
```
Comment (SylviaBABY) [inline docs/zh/latest/plugins/fault-injection.md:52]:
```suggestion
- `abort` 属性将直接返回给客户端指定的响应码并且终止其他插件的执行。
- `delay` 属性将延迟某个请求，并且还会执行配置的其他插件。
- `abort` 和 `delay` 属性至少要配置一个。
```

File: docs/zh/latest/plugins/fault-injection.md [hunk 4]
```diff
@@ -23,23 +29,33 @@ title: fault-injection
...
 
-注：参数 abort 和 delay 至少要存在一个。
+:::
 
-`vars` 是由 `lua-resty-expr` 的表达式组成的列表，它可以灵活的实现规则之间的 `and/or` 关系，示例：
+:::tip
+
+`vars` 是由 [`lua-resty-expr`](https://github.com/api7/lua-resty-expr) 的表达式组成的列表，它可以灵活的实现规则之间的 AND/OR 关系，示例如下：
```
Comment (SylviaBABY) [inline docs/zh/latest/plugins/fault-injection.md:58]:
```suggestion
`vars` 是由 [`lua-resty-expr`](https://github.com/api7/lua-resty-expr) 的表达式组成的列表，它可以灵活地实现规则之间的 AND/OR 关系，示例如下：
```

File: docs/zh/latest/plugins/fault-injection.md [hunk 5]
```diff
@@ -298,99 +159,65 @@ curl http://127.0.0.1:9080/apisix/admin/routes/1  -H 'X-API-KEY: edd1c9f034335f1
 }'
 ```
 
-测试：
+## 测试插件
 
-1、abort 和 delay 的 vars 规则匹配失败：
+通过上述示例启用插件后，可以向路由发起请求：
```
Comment (SylviaBABY) [inline docs/zh/latest/plugins/fault-injection.md:164]:
```suggestion
通过上述示例启用插件后，可以向路由发起如下请求：
```

File: docs/zh/latest/plugins/grpc-transcode.md [hunk 6]
```diff
@@ -23,20 +29,40 @@ title: grpc-transcode
 
 ## 描述
 
-HTTP(s) -> APISIX -> gRPC server
+`grpc-transcode` 插件可以在 HTTP 和 gRPC 请求之间进行转换。
```
Comment (SylviaBABY) [inline docs/zh/latest/plugins/grpc-transcode.md:32]:
```suggestion
使用 `grpc-transcode` 插件可以在 HTTP 和 gRPC 请求之间进行转换。
```

File: docs/zh/latest/plugins/grpc-transcode.md [hunk 7]
```diff
@@ -23,20 +29,40 @@ title: grpc-transcode
...
 
-## Proto
+APISIX 接收 HTTP 请求后，首先对请求进行转码，并将转码后的请求转发到 gRPC 服务，获取响应并以 HTTP 格式将其返回给客户端。
 
-### 参数
+<!-- TODO: use an image here to explain the concept better -->
```
Comment (SylviaBABY) [inline docs/zh/latest/plugins/grpc-transcode.md:36]: you can use picture in this post:https://apisix.apache.org/zh/blog/2022/01/25/apisix-grpc-web-integration

File: docs/zh/latest/plugins/grpc-transcode.md [hunk 8]
```diff
@@ -75,14 +100,15 @@ protoc --include_imports --descriptor_set_out=proto.pb proto/helloworld.proto
 
 然后我们可以将 `proto.pb` 的内容作为 proto 的 `content` 字段提交。
```
Comment (SylviaBABY) [inline docs/zh/latest/plugins/grpc-transcode.md:101]:
```suggestion
然后将 `proto.pb` 的内容作为 proto 的 `content` 字段提交。
```

File: docs/zh/latest/plugins/grpc-transcode.md [hunk 9]
```diff
@@ -75,14 +100,15 @@ protoc --include_imports --descriptor_set_out=proto.pb proto/helloworld.proto
...
 
-由于内容是二进制的，我们需要先对其进行 base64 编码。这里我们用一个 Python 脚本来做。
+由于 proto 的内容是二进制的，我们需要使用以下 Python 脚本将其转换成 `base64`：
```
Comment (SylviaBABY) [inline docs/zh/latest/plugins/grpc-transcode.md:103]:
```suggestion
由于 proto 的内容是二进制的，我们需要使用以下 Python 脚本将其转换成 `base64` 格式：
```

File: docs/zh/latest/plugins/grpc-web.md [hunk 10]
```diff
@@ -50,19 +53,35 @@ curl http://127.0.0.1:9080/apisix/admin/routes/1 -H 'X-API-KEY: edd1c9f034335f13
 }'
 ```
 
+:::info IMPORTANT
+
+在使用 `gRPC Web` 代理插件时，路由必须使用**前缀匹配**模式（例如：`/*` 或 `/grpc/example/*`），因为 `gRPC Web` 客户端会在 URI 中传递 `proto` 中声明的**包名称**、**服务接口名称**、**方法名称**等信息（例如：`/path/a6.RouteService/Insert`）,
+
+因此在使用**绝对匹配**时将无法命中插件和提取 `proto` 信息。
```
Comment (SylviaBABY) [inline docs/zh/latest/plugins/grpc-web.md:60]:
```suggestion
在使用 `gRPC Web` 代理插件时，路由必须使用**前缀匹配**模式（例如：`/*` 或 `/grpc/example/*`），因为 `gRPC Web` 客户端会在 URI 中传递 `proto` 中声明的**包名称**、**服务接口名称**、**方法名称**等信息（例如：`/path/a6.RouteService/Insert`），因此在使用**绝对匹配**时将无法命中插件和提取 `proto` 信息。
```

File: docs/zh/latest/plugins/mocking.md [hunk 11]
```diff
@@ -23,22 +29,29 @@ title: mocking
 
 ## 描述
 
-Mock API 插件，绑定该插件后将随机返回指定格式的`mock`数据，不再转发到后端。
+`mocking` 插件用于 Mock API。当执行该插件时，它将随机返回指定格式的模拟数据，并且请求不会转发到上游。
```
Comment (SylviaBABY) [inline docs/zh/latest/plugins/mocking.md:32]:
```suggestion
`mocking` 插件用于处理 Mock API 相关。当执行该插件时，它将随机返回指定格式的模拟数据，并且请求不会转发到上游。
```

File: docs/zh/latest/plugins/mocking.md [hunk 12]
```diff
@@ -23,22 +29,29 @@ title: mocking
...
+| response_status | integer| 否    | 200              | 返回响应的 HTTP 状态码。                                            |
+| content_type    | string | 否    | application/json | 返回响应的 Header `Content-Type`。                                            |
+| response_example| string | 否    |                  | 返回响应的 Body，与 `response_schema` 字段二选一。                                            |
+| response_schema | object | 否    |                  | 指定响应的 `jsonschema` 对象，未指定 `response_example` 字段时生效。                        |
+| with_mock_header| boolean| 否    | true             | 当设置为 `true` 时，将添加响应头 `x-mock-by: APISIX/{version}`。设置为 `false` 时则不添加该响应头。   |
 
-支持的字段类型：`string`, `number`, `integer`, `boolean`, `object`, `array`
-基础数据类型（`string`,`number`,`integer`,`boolean`）可通过配置`example`属性指定生成的响应值，未配置时随机返回。
-以下是一个`jsonschema`实例：
+JSON schema 在其字段中支持以下类型：
```
Comment (SylviaBABY) [inline docs/zh/latest/plugins/mocking.md:45]:
```suggestion
JSON Schema 在其字段中支持以下类型：
```

File: docs/zh/latest/plugins/mocking.md [hunk 13]
```diff
@@ -23,22 +29,29 @@ title: mocking
...
+
+- `string`
+- `number`
+- `integer`
+- `boolean`
+- `object`
+- `array`
+
+以下是一个 JSON schema 示例：
```
Comment (SylviaBABY) [inline docs/zh/latest/plugins/mocking.md:54]:
```suggestion
以下是一个 JSON Schema 示例：
```

File: docs/zh/latest/plugins/mocking.md [hunk 14]
```diff
@@ -86,7 +99,7 @@ Mock API 插件，绑定该插件后将随机返回指定格式的`mock`数据
 }
 ```
 
-以下为该`jsonschema`可能生成的返回对象：
+以下为上述 JSON schema 可能生成的返回对象：
```
Comment (SylviaBABY) [inline docs/zh/latest/plugins/mocking.md:102]:
```suggestion
以下为上述 JSON Schema 可能生成的返回对象：
```

File: docs/zh/latest/plugins/proxy-rewrite.md [hunk 15]
```diff
@@ -23,25 +29,26 @@ title: proxy-rewrite
 
 ## 描述
 
-proxy-rewrite 是上游代理信息重写插件，支持对 `scheme`、`uri`、`host` 等信息的重写。
+`proxy-rewrite` 是上游代理信息重写插件，支持对 `scheme`、`uri`、`host` 等信息的重写。
```
Comment (SylviaBABY) [inline docs/zh/latest/plugins/proxy-rewrite.md:32]:
```suggestion
`proxy-rewrite` 是处理上游代理信息重写的插件，支持对 `scheme`、`uri`、`host` 等信息进行重写。
```

File: docs/zh/latest/plugins/response-rewrite.md [hunk 16]
```diff
@@ -103,12 +119,23 @@ X-Server-balancer_addr: 127.0.0.1:80
 {"code":"ok","message":"new json body"}
 ```
 
-### 禁用插件
+:::info IMPORTANT
+
+[ngx.exit](https://openresty-reference.readthedocs.io/en/latest/Lua_Nginx_API/#ngxexit) 将会中断当前请求的执行并将其状态码返回给 NGINX。
+
+如果你在 `access` 阶段执行 `ngx.exit`，只是中断了请求处理阶段，响应阶段仍然会处理。比如，如果你配置了 `response-rewrite` 插件，它会强制覆盖你的响应信息（如响应代码）。
```
Comment (SylviaBABY) [inline docs/zh/latest/plugins/response-rewrite.md:126]:
```suggestion
如果你在 `access` 阶段执行 `ngx.exit`，只是中断了请求处理阶段，响应阶段仍然会处理。如果你配置了 `response-rewrite` 插件，它会强制覆盖你的响应信息（如响应代码）。
```

**Review by SylviaBABY — DISMISSED** [review]

**Comment by SylviaBABY** [comment]
LGTM

File: docs/zh/latest/plugins/response-rewrite.md [hunk 17]
```diff
@@ -23,39 +29,47 @@ title: response-rewrite
...
+| filters.regex   | string  | 是  |        |                 | 用于匹配响应体正则表达式。                                                                                                               |
+| filters.scope   | string  | 否  | "once" | "once","global" | 替换范围，"once" 表达式 `filters.regex` 仅替换首次匹配上响应体的内容，"global" 则进行全局替换。                                                                                               |
+| filters.replace | string  | 是  |        |                 | 替换后的内容。                                                                                                                                                        |
+| filters.options | string  | 否  | "jo"   |                 | 正则匹配有效参数，可选项见 [ngx.re.match](https://github.com/openresty/lua-nginx-module#ngxrematch)。                                                                                                           |
 
-`body` 和 `filters`，两个只能配置其中一个。
+:::note
 
-## 示例
+`body` 和 `filters`属性只能配置其中一个。
```
Comment (juzhiyuan) [inline docs/zh/latest/plugins/response-rewrite.md:62]:
```suggestion
`body` 和 `filters` 属性只能配置其中一个。
```

**Review by juzhiyuan — APPROVED** [review]
LGTM

File: docs/zh/latest/plugins/grpc-transcode.md [hunk 18]
```diff
@@ -23,20 +29,40 @@ title: grpc-transcode
...
 ## 描述
 
-HTTP(s) -> APISIX -> gRPC server
+使用 `grpc-transcode` 插件可以在 HTTP 和 gRPC 请求之间进行转换。
 
-## Proto
+APISIX 接收 HTTP 请求后，首先对请求进行转码，并将转码后的请求转发到 gRPC 服务，获取响应并以 HTTP 格式将其返回给客户端。
 
-### 参数
+<!-- TODO: use an image here to explain the concept better -->
```
[the remark this answers is not in the document]
Comment (hf400159) [reply docs/zh/latest/plugins/grpc-transcode.md:36]: No, I have no idea. but I'll do it later.

**Review by SylviaBABY — APPROVED** [review]

**Review by juzhiyuan — APPROVED** [review]

**Pull request merged**